numpy/lib/tests/test_function_base.py

```diff
@@ -1084,6 +1084,18 @@ def test_values(self):
         assert_raises(ValueError, gradient, np.arange(1), edge_order=2)
         assert_raises(ValueError, gradient, np.arange(2), edge_order=2)
 
+    @pytest.mark.parametrize('f_dtype', [np.uint8, np.uint32])
+    def test_f_decreasing_unsigned_int(self, f_dtype):
+        f = np.array([5, 4, 3, 2, 1], dtype=f_dtype)
+        g = gradient(f)
+        assert_array_equal(g, np.full(len(f), fill_value=-1))
```

Comment:
is there a reason to limit this to only these dtypes? Perhaps it should be testing as many as possible (not datetime64 or object)

Reply:
I pushed an update with more thorough tests for the integer types.